### PR TITLE
Fix robot.txt and generate new one when action on languages

### LINKED ISSUE #14836: robots.txt is wrong
**Describe the bug**
Robots.txt file is generated for wrong dirs.

**To Reproduce**
0. Install Danish language on a clean install (no second language). Try to install PrestaShop in a folder instead of in the root (I did not test in the root).
1. Go to configure -> shop -> seo & url -> generate robots.txt
2. See robots.txt Should be similair to below:
**Please note that there is generated some URLs with language code `/da/` but I do only have one language installed, so I don't have a language code in my URL.**

```
# robots.txt automatically generated by PrestaShop e-commerce open-source solution
# http://www.prestashop.com - http://www.prestashop.com/forums
# This file is to prevent the crawling and indexing of certain parts
# of your site by web crawlers and spiders run by sites like Yahoo!
# and Google. By telling these "robots" where not to go on your site,
# you save bandwidth and server resources.
# For more information about the robots.txt standard, see:
# http://www.robotstxt.org/robotstxt.html
User-agent: *
# Allow Directives
Allow: */modules/*.css
Allow: */modules/*.js
Allow: */modules/*.png
Allow: */modules/*.jpg
Allow: /js/jquery/*
# Private pages
Disallow: /*?order=
Disallow: /*?tag=
Disallow: /*?id_currency=
Disallow: /*?search_query=
Disallow: /*?back=
Disallow: /*?n=
Disallow: /*&order=
Disallow: /*&tag=
Disallow: /*&id_currency=
Disallow: /*&search_query=
Disallow: /*&back=
Disallow: /*&n=
Disallow: /*controller=addresses
Disallow: /*controller=address
Disallow: /*controller=authentication
Disallow: /*controller=cart
Disallow: /*controller=discount
Disallow: /*controller=footer
Disallow: /*controller=get-file
Disallow: /*controller=header
Disallow: /*controller=history
Disallow: /*controller=identity
Disallow: /*controller=images.inc
Disallow: /*controller=init
Disallow: /*controller=my-account
Disallow: /*controller=order
Disallow: /*controller=order-slip
Disallow: /*controller=order-detail
Disallow: /*controller=order-follow
Disallow: /*controller=order-return
Disallow: /*controller=order-confirmation
Disallow: /*controller=pagination
Disallow: /*controller=password
Disallow: /*controller=pdf-invoice
Disallow: /*controller=pdf-order-return
Disallow: /*controller=pdf-order-slip
Disallow: /*controller=product-sort
Disallow: /*controller=search
Disallow: /*controller=statistics
Disallow: /*controller=attachment
Disallow: /*controller=guest-tracking
# Directories for localhost
Disallow: /prestashop_WH/app/
Disallow: /prestashop_WH/cache/
Disallow: /prestashop_WH/classes/
Disallow: /prestashop_WH/config/
Disallow: /prestashop_WH/controllers/
Disallow: /prestashop_WH/download/
Disallow: /prestashop_WH/js/
Disallow: /prestashop_WH/localization/
Disallow: /prestashop_WH/log/
Disallow: /prestashop_WH/mails/
Disallow: /prestashop_WH/modules/
Disallow: /prestashop_WH/override/
Disallow: /prestashop_WH/pdf/
Disallow: /prestashop_WH/src/
Disallow: /prestashop_WH/tools/
Disallow: /prestashop_WH/translations/
Disallow: /prestashop_WH/upload/
Disallow: /prestashop_WH/var/
Disallow: /prestashop_WH/vendor/
Disallow: /prestashop_WH/webservice/
Disallow: /da/app/
Disallow: /da/cache/
Disallow: /da/classes/
Disallow: /da/config/
Disallow: /da/controllers/
Disallow: /da/download/
Disallow: /da/js/
Disallow: /da/localization/
Disallow: /da/log/
Disallow: /da/mails/
Disallow: /da/modules/
Disallow: /da/override/
Disallow: /da/pdf/
Disallow: /da/src/
Disallow: /da/tools/
Disallow: /da/translations/
Disallow: /da/upload/
Disallow: /da/var/
Disallow: /da/vendor/
Disallow: /da/webservice/
# Files
Disallow: /*da/gendan-adgangskode
Disallow: /*da/adresse
Disallow: /*da/adresser
Disallow: /*da/log-ind
Disallow: /*da/kurv
Disallow: /*da/rabat
Disallow: /*da/ordrehistorik
Disallow: /*da/mine-oplysninger
Disallow: /*da/min-konto
Disallow: /*da/foelg-din-bestilling
Disallow: /*da/kreditnota
Disallow: /*da/ordre
Disallow: /*da/soeg
Disallow: /*da/Gaestesporing
Disallow: /*da/ordrebekraeftelse

```

**Additional information**
PrestaShop version: 1.7.6.0
PHP version: 7.2.x
screen

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Currently the robots.txt is not good when there is only one language. I've added an improvement to generate the robots.txt when you change a language. I also add svg and gif
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/14836
| How to test?      | See below
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

**How to test :** 

- Generate robots.txt with one language and the url is http://127.0.0.1/  and check the file
- Generate robots.txt with two languages and the url is http://127.0.0.1/  and check the file
- Generate robots.txt with one language and the url is http://127.0.0.1/myfolder/  and check the file
- Generate robots.txt with two languages and and the url is http://127.0.0.1/myfolder/  and check the file
- Put an online website and check if the crawler dont acces to the page with this tools (https://fr.ryte.com/free-tools/robots-txt/) or another tools

The new content without /fr/ or /*fr/ for a website with one language and in the folder `/dev-monoshop/PrestaShop/`

```
# robots.txt automatically generated by PrestaShop e-commerce open-source solution
# https://www.prestashop.com - https://www.prestashop.com/forums
# This file is to prevent the crawling and indexing of certain parts
# of your site by web crawlers and spiders run by sites like Yahoo!
# and Google. By telling these "robots" where not to go on your site,
# you save bandwidth and server resources.
# For more information about the robots.txt standard, see:
# https://www.robotstxt.org/robotstxt.html
User-agent: *
# Allow Directives
Allow: */modules/*.css
Allow: */modules/*.js
Allow: */modules/*.png
Allow: */modules/*.jpg
Allow: */modules/*.gif
Allow: */modules/*.svg
Allow: /js/jquery/*
# Private pages
Disallow: /*?order=
Disallow: /*?tag=
Disallow: /*?id_currency=
Disallow: /*?search_query=
Disallow: /*?back=
Disallow: /*?n=
Disallow: /*&order=
Disallow: /*&tag=
Disallow: /*&id_currency=
Disallow: /*&search_query=
Disallow: /*&back=
Disallow: /*&n=
Disallow: /*controller=addresses
Disallow: /*controller=address
Disallow: /*controller=authentication
Disallow: /*controller=cart
Disallow: /*controller=discount
Disallow: /*controller=footer
Disallow: /*controller=get-file
Disallow: /*controller=header
Disallow: /*controller=history
Disallow: /*controller=identity
Disallow: /*controller=images.inc
Disallow: /*controller=init
Disallow: /*controller=my-account
Disallow: /*controller=order
Disallow: /*controller=order-slip
Disallow: /*controller=order-detail
Disallow: /*controller=order-follow
Disallow: /*controller=order-return
Disallow: /*controller=order-confirmation
Disallow: /*controller=pagination
Disallow: /*controller=password
Disallow: /*controller=pdf-invoice
Disallow: /*controller=pdf-order-return
Disallow: /*controller=pdf-order-slip
Disallow: /*controller=product-sort
Disallow: /*controller=search
Disallow: /*controller=statistics
Disallow: /*controller=attachment
Disallow: /*controller=guest-tracking
# Directories for 127.0.0.1
Disallow: /dev-monoshop/PrestaShop/app/
Disallow: /dev-monoshop/PrestaShop/cache/
Disallow: /dev-monoshop/PrestaShop/classes/
Disallow: /dev-monoshop/PrestaShop/config/
Disallow: /dev-monoshop/PrestaShop/controllers/
Disallow: /dev-monoshop/PrestaShop/download/
Disallow: /dev-monoshop/PrestaShop/js/
Disallow: /dev-monoshop/PrestaShop/localization/
Disallow: /dev-monoshop/PrestaShop/log/
Disallow: /dev-monoshop/PrestaShop/mails/
Disallow: /dev-monoshop/PrestaShop/modules/
Disallow: /dev-monoshop/PrestaShop/override/
Disallow: /dev-monoshop/PrestaShop/pdf/
Disallow: /dev-monoshop/PrestaShop/src/
Disallow: /dev-monoshop/PrestaShop/tools/
Disallow: /dev-monoshop/PrestaShop/translations/
Disallow: /dev-monoshop/PrestaShop/upload/
Disallow: /dev-monoshop/PrestaShop/var/
Disallow: /dev-monoshop/PrestaShop/vendor/
Disallow: /dev-monoshop/PrestaShop/webservice/
# Files
Disallow: /dev-monoshop/PrestaShop/recuperation-mot-de-passe
Disallow: /dev-monoshop/PrestaShop/adresse
Disallow: /dev-monoshop/PrestaShop/adresses
Disallow: /dev-monoshop/PrestaShop/connexion
Disallow: /dev-monoshop/PrestaShop/panier
Disallow: /dev-monoshop/PrestaShop/reduction
Disallow: /dev-monoshop/PrestaShop/historique-commandes
Disallow: /dev-monoshop/PrestaShop/identite
Disallow: /dev-monoshop/PrestaShop/mon-compte
Disallow: /dev-monoshop/PrestaShop/suivi-commande
Disallow: /dev-monoshop/PrestaShop/avoirs
Disallow: /dev-monoshop/PrestaShop/commande
Disallow: /dev-monoshop/PrestaShop/recherche
Disallow: /dev-monoshop/PrestaShop/suivi-commande-invite
Disallow: /dev-monoshop/PrestaShop/confirmation-commande
```

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26133)

### CODE REVIEW

File: classes/Tools.php [hunk 1]
```diff
@@ -2950,6 +2951,8 @@ public static function getRobotsContent()
             '*/modules/*.js',
             '*/modules/*.png',
             '*/modules/*.jpg',
+            '*/modules/*.gif',
+            '*/modules/*.svg',
```
Comment (Progi1984) [inline classes/Tools.php:2955]:
```suggestion
            '*/modules/*.svg',
            '*/modules/*.webp',
```

File: src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php [hunk 2]
```diff
@@ -130,6 +130,9 @@ public function createAction(Request $request)
             if (null !== $result->getIdentifiableObjectId()) {
                 $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
 
+                $robotsTextFileGenerator = $this->get('prestashop.adapter.file.robots_text_file_generator');
+                $robotsTextFileGenerator->generateFile();
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php]: May be could you move it to a private method ?

**Review by Progi1984 — COMMENTED** [review]
A small comment

File: src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php (hunk 2 above)
Comment (ghost) [reply src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php]: I will do but on Monday

**Pull request closed without merging**